Source/Core/DolphinQt/MainWindow.cpp

```diff
@@ -1,362 +1,297 @@
-			m_render_widget->setMinimumSize(0, 0); // Allow the widget to scale down
-		}
-		m_render_widget->adjustSize();
-	}
+	set_table->setCheckable(true);
+	set_table->setChecked(true);
+	set_list->setCheckable(true);
 
-	if (!BootManager::BootCore(filename.toStdString()))
-	{
-		QMessageBox::critical(this, tr("Fatal error"), tr("Failed to init Core"), QMessageBox::Ok);
-		if (SConfig::GetInstance().bFullscreen)
-			m_render_widget->close();
-		else
-			m_ui->centralWidget->removeWidget(m_render_widget.get());
-		m_render_widget.reset();
-	}
-	else
-	{
-		DisableScreensaver();
-		emit CoreStateChanged(Core::CORE_RUN);
-	}
+	connect(set_table, &QAction::triggered, m_game_list, &GameList::SetTableView);
+	connect(set_list, &QAction::triggered, m_game_list, &GameList::SetListView);
+
+	menuBar()->addMenu(tr("Help"));
 }
 
-void DMainWindow::DisableScreensaver()
+void MainWindow::MakeToolBar()
 {
-#ifdef Q_OS_WIN
-	// Prevents Windows from sleeping or turning off the display
-	SetThreadExecutionState(ES_CONTINUOUS | ES_DISPLAY_REQUIRED);
-#endif
+	setToolButtonStyle(Qt::ToolButtonTextUnderIcon);
+	m_tool_bar = addToolBar(tr("ToolBar"));
+	m_tool_bar->setMovable(false);
+	m_tool_bar->setFloatable(false);
+
+	PopulateToolBar();
 }
 
-void DMainWindow::EnableScreensaver()
+void MainWindow::MakeGameList()
 {
-#ifdef Q_OS_WIN
-	// Allows Windows to sleep and turn off the display
-	SetThreadExecutionState(ES_CONTINUOUS);
-#endif
+	m_game_list = new GameList(this);
+	connect(m_game_list, &GameList::GameSelected, this, &MainWindow::Play);
 }
 
-QString DMainWindow::RequestBootFilename()
+void MainWindow::MakeRenderWidget()
 {
-	// If a game is already selected, just return the filename
-	if (m_game_tracker->SelectedGame() != nullptr)
-			return m_game_tracker->SelectedGame()->GetFileName();
-
-	return ShowFileDialog();
+	m_render_widget = new RenderWidget;
+	connect(m_render_widget, &RenderWidget::EscapePressed, this, &MainWindow::Stop);
+	m_render_widget->hide();
+	m_rendering_to_main = false;
 }
 
-QString DMainWindow::ShowFileDialog()
+void MainWindow::MakeStack()
 {
-	return QFileDialog::getOpenFileName(this, tr("Open File"), QString(),
-		tr("All supported ROMs (%1);;All files (*)")
-		.arg(SL("*.gcm *.iso *.ciso *.gcz *.wbfs *.elf *.dol *.dff *.tmd *.wad")));
+	m_stack = new QStackedWidget;
+	m_stack->setMinimumSize(800, 600);
+	m_stack->addWidget(m_game_list);
+	setCentralWidget(m_stack);
 }
 
-QString DMainWindow::ShowFolderDialog()
+void MainWindow::PopulateToolBar()
 {
-	return QFileDialog::getExistingDirectory(this, tr("Browse for a directory to add"),
-	                                         QDir::homePath(),
-	                                         QFileDialog::ShowDirsOnly);
+	m_tool_bar->clear();
+
+	QString dir = QString::fromStdString(File::GetThemeDir(SConfig::GetInstance().theme_name));
+	m_tool_bar->addAction(QIcon(dir + tr("open.png")), tr("Open"), this, SLOT(Open()));
```

Comment:
I don't think it makes sense to allow editing the string that composes the filepath.